src/libraries/sfx-finish.proj

```diff
@@ -0,0 +1,129 @@
+    </ItemGroup>
+
+    <CreateFrameworkListFile Files="@(FrameworkListInputFile)"
+                             TargetFile="$(FrameworkListOutputPath)"
+                             TargetFilePrefixes="ref/;runtimes/;analyzers/"
+                             RootAttributes="@(FrameworkListRootAttribute)" />
+  </Target>
+
+  <!-- Copy the intellisense file to the targeting pack directory. -->
+  <Target Name="CopyDocumentationFiles" AfterTargets="Build">
+    <ItemGroup>
+      <DocumentationFileToCopy Include="@(SharedFrameworkAssembly->WithMetadataValue('IsPrivateAssembly', 'false')->Metadata('DocFileItem'))" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(DocumentationFileToCopy)"
+          DestinationFolder="$(MicrosoftNetCoreAppRefPackRefDir)"
+          SkipUnchangedFiles="true"
+          UseHardlinksIfPossible="true" />
+  </Target>
+
+  <Target Name="ValidateSharedFramework" AfterTargets="Build">
+    <ItemGroup>
+      <!-- exclude private assemblies from ref-->
+      <_expectedRef Include="@(NetCoreAppLibrary)" Condition="!$([System.String]::new('%(Identity)').StartsWith('System.Private'))" />
+      <_expectedSharedFrameworkFile Include="@(_expectedRefs->'$(MicrosoftNetCoreAppRefPackRefDir)%(Identity).dll')" />
+
+      <!-- exclude the full facades from expected docs since they have no types -->
+      <_expectedDoc Include="@(_expectedRefs)" Exclude="netstandard;@(NetFxReference)" />
+      <_expectedSharedFrameworkFile Include="@(_expectedDoc->'$(MicrosoftNetCoreAppRefPackRefDir)%(Identity).xml')" />
+
+      <!-- exclude CoreLib from expected libs, since it's placed in native -->
+      <_expectedLib Include="@(NetCoreAppLibrary)" Exclude="System.Private.CoreLib" />
+      <_expectedSharedFrameworkFile Include="@(_expectedLib->'$(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)%(Identity).dll')" />
+
+      <_missingSharedFrameworkFile Include="@(_expectedSharedFrameworkFile)" Exclude="@(_expectedSharedFrameworkFile->Exists())" />
+    </ItemGroup>
+
+    <Error Text="The shared framework files '@(_missingSharedFrameworkFile)' were missing." Condition="'@(_missingSharedFrameworkFile)' != ''" />
+  </Target>
+
+  <Target Name="GetTrimSharedFrameworkAssembliesInputs" DependsOnTargets="ResolveProjectReferences">
+    <PropertyGroup>
+      <SharedFrameworkAssembliesTrimDir>$([MSBuild]::NormalizeDirectory('$(IntermediateOutputPath)', 'illink-$(TargetArchitecture)'))</SharedFrameworkAssembliesTrimDir>
+      <SharedFrameworkAssembliesTrimMarkerFile>$(SharedFrameworkAssembliesTrimDir)linker-marker.txt</SharedFrameworkAssembliesTrimMarkerFile>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <SharedFrameworkSuppressionsXml Include="@(SharedFrameworkAssembly->Metadata('ILLinkSuppressionsXmls'))" />
+      <SharedFrameworkSuppressionsXml Include="$(CoreLibSharedDir)ILLink\ILLink.Suppressions.LibraryBuild.xml" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="TrimSharedFrameworkAssemblies"
```

Comment:
Would it be useful to add trimming support to the sfxproj infrastructure? I don't know if any of the other shared frameworks would find it useful to be trimmed in library mode before being r2r'd.

Reply:
This target trims the entire shared framework libraries in library mode for the purpose of validation. We don't use the outputs anywhere. I wasn't involved in adding this validation step so I'm not sure how useful it is or if we still need it. @ericstj might know.

Reply:
Looks to me this might have been first introduced in https://github.com/dotnet/runtime/pull/40172 as part of https://github.com/dotnet/runtime/issues/38033.  I think the goal there would be to catch any sort of linker warnings that might exist before our customers do.  cc @eerhardt

Reply:
Correct - this is to validate the trimmer warnings in the whole shared framework. It doesn't affect the output.

The ASP.NET shared framework does something similar as well. 

https://github.com/dotnet/aspnetcore/blob/67ae6c8b73e80fa5f499a069443c9844ca86c62a/src/Tools/LinkabilityChecker/LinkabilityChecker.csproj#L21